src/libsyntax/codemap.rs

```diff
@@ -598,27 +605,27 @@ impl CodeMap {
         Ok(self.new_filemap(path.to_str().unwrap().to_string(), src))
     }
 
+    fn next_start_pos(&self) -> usize {
+        let files = self.files.borrow();
+        match files.last() {
+            None => 0,
```

Comment:
This is a good idea, but I couldn't make it work :-( Basically there are a bunch of cases where we have a DUMMY_SP and try and treat it like a proper one. Might be interesting to fix this all properly, but I don't have time.